docs/modules/deploy/pages/deploying-in-kubernetes.adoc

```diff
@@ -50,3 +59,401 @@ Since all the Operators mentioned above are generated from Helm Charts, their fu
+
+NOTE: Your Kubernetes cluster must orchestrate Hazelcast Member Pods equally between the availability zones, otherwise Zone Aware feature may not work correctly.
+
+[tabs] 
+==== 
+YAML:: 
++ 
+-- 
+```yaml
+partition-group:
+  enabled: true
+  group-type: ZONE_AWARE
+```
+--
+Java:: 
++ 
+-- 
+```java
+config.getPartitionGroupConfig()
+    .setEnabled(true)
+    .setGroupType(MemberGroupType.ZONE_AWARE);
+```
+--
+====
+
+Note the following aspects of `ZONE_AWARE`:
+
+* Kubernetes cluster must provide the link:https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/#failure-domainbetakubernetesiozone[well-known Kubernetes annotations]
+* Retrieving Zone Name uses Kubernetes API, so <<granting-permissions-to-use-kubernetes-api, RBAC must be configured>>
+* `ZONE_AWARE` feature works correctly when Hazelcast members are distributed equally in all zones, so your Kubernetes cluster must orchestrate Pods equally
+ 
+ Note also that retrieving Zone Name assumes that your container's hostname is the same as Pod Name, which is almost always true. If you happen to change your hostname in the container, then please define the following environment variable:
+ 
+```yaml
+env:
+  - name: POD_NAME
+    valueFrom:
+      fieldRef:
+        fieldPath: metadata.name
+``` 
+
+=== Node Aware
+
+When using `NODE_AWARE` configuration, backups are created in the other Kubernetes nodes. This feature is available only for the Kubernetes API mode.
+
+NOTE: Your Kubernetes cluster must orchestrate Hazelcast Member Pods equally between the nodes, otherwise Node Aware feature may not work correctly.
+
+==== YAML Configuration
+
+```yaml
+partition-group:
+  enabled: true
+  group-type: NODE_AWARE
+```
+
+==== Java-based Configuration
+
+```java
+config.getPartitionGroupConfig()
+    .setEnabled(true)
+    .setGroupType(MemberGroupType.NODE_AWARE);
+```
+
+Note the following aspects of `NODE_AWARE`:
+
+* Retrieving name of the node uses Kubernetes API, so <<granting-permissions-to-use-kubernetes-api, RBAC must be configured>>
+* `NODE_AWARE` feature works correctly when Hazelcast members are distributed equally in all nodes, so your Kubernetes cluster must orchestrate Pods equally.
+ 
+Note also that retrieving name of the node assumes that your container's hostname is the same as Pod Name, which is almost always true. If you happen to change your hostname in the container, then please define the following environment variable:
+ 
+ ```yaml
+env:
+  - name: POD_NAME
+    valueFrom:
+      fieldRef:
+        fieldPath: metadata.name
+ ``` 
+
+== Discovering Members from Hazelcast Clients
+
+Depending on whether your Hazelcast client runs inside or outside the Kubernetes cluster, its configuration looks different.
+
+=== Inside Kubernetes Cluster
+
+If you have a Hazelcast cluster deployed on Kubernetes and you want to configure Hazelcast Client deployed on the same Kubernetes cluster, then the configuration looks very similar to Hazelcast Member.
```

Comment:
I think, in the other parts of documentation, we generally use "Hazelcast client" or "Hazelcast member" instead of "Hazelcast Client" or "Hazelcast Member".